subprojects/language-groovy/src/main/java/org/gradle/api/tasks/compile/GroovyCompile.java

```diff
@@ -303,11 +307,50 @@ private GroovyJavaJointCompileSpec createSpec() {
             GFileUtils.mkdirs(dir);
             spec.getGroovyCompileOptions().setStubDir(dir);
         }
-        spec.getCompileOptions().getForkOptions().setExecutable(Jvm.current().getJavaExecutable().getAbsolutePath());
+
+        if (javaLauncher.isPresent()) {
+            applyToolchain(spec.getCompileOptions().getForkOptions());
+        } else {
+            spec.getCompileOptions().getForkOptions().setExecutable(Jvm.current().getJavaExecutable().getAbsolutePath());
+        }
 
         return spec;
     }
 
+    private void configureCompatibilityOptions(DefaultGroovyJavaJointCompileSpec spec) {
+        JavaInstallationMetadata toolchain = getToolchain();
+        if (toolchain != null) {
+            boolean isSourceOrTargetConfigured = false;
+            if (super.getSourceCompatibility() != null) {
+                spec.setSourceCompatibility(getSourceCompatibility());
+                isSourceOrTargetConfigured = true;
+            }
+            if (super.getTargetCompatibility() != null) {
+                spec.setTargetCompatibility(getTargetCompatibility());
+                isSourceOrTargetConfigured = true;
+            }
+            if (!isSourceOrTargetConfigured) {
+                String languageVersion = toolchain.getLanguageVersion().toString();
```

Comment:
`JavaCompile` takes `release` into account as well depending on the toolchain, I think that might be worth here as well as the spec supports release already. 

https://github.com/gradle/gradle/blob/41ccab532c7c2724796c996ed8884155982ee5a1/subprojects/language-java/src/main/java/org/gradle/api/tasks/compile/JavaCompile.java#L417

Reply:
The spec supports it because of reuse, but last I checked Groovy did not. And I did not want to include such a change in this context tbh.